service/src/test/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiControllerConnectedTest.java

```diff
@@ -88,6 +100,29 @@ public void createBigQueryDataset_createdResourceEqualsGotResource() throws Exce
         actualBqDataset.getAttributes().getProjectId());
   }
 
+  @Test
```

Comment:
*optional* - I have come around to Zach's thinking that for tests like this, where you don't need real GCP, unit test is better. You don't have to wait for calls to real services or for GCP project to be created, so test should be faster. Put this in ControlledGcpResourceApiControllerTest.java, which is unit test. Same for rest of PR.

Reply:
moved. the other test is in unit.